### PR TITLE
Add upsample operator

### PR DESCRIPTION
**Pull request by joddiy**

### CODE REVIEW

File: python/singa/autograd.py [hunk 1]
```diff
@@ -4698,6 +4698,82 @@ def cossim(a, b):
     return CosSim()(a, b)[0]
 
 
+class UpSample(Operator):
+    """
+    Init a UpSample operator
```
Comment (nudles) [inline python/singa/autograd.py:4703]: why use "Init" here?

File: python/singa/autograd.py [hunk 2]
```diff
@@ -4698,6 +4698,82 @@ def cossim(a, b):
...
+    def __init__(self, mode, scales):
+        """
+        Args:
+            scales (list[int]): The scale array along each dimension. It takes 
+                value greater than or equal to 1. 
+        """
+        super(UpSample, self).__init__()
+        self.scales = scales
+        self.mode = mode
+        if self.mode != "nearest":
```
Comment (nudles) [inline python/singa/autograd.py:4715]: better use self.mode.lower() != "nearest"

File: python/singa/autograd.py [hunk 3]
```diff
@@ -4698,6 +4698,82 @@ def cossim(a, b):
...
+        for axis, s in zip(range(len(self.scales)), self.scales):
+            s = int(s)
+            if s == 1:
+                continue
+            x = x.Repeat([s, ], axis)
+        return x
+
+    def backward(self, dy):
+        """
+        backward of UpSample
```
Comment (nudles) [inline python/singa/autograd.py:4740]: this kind of docstring is trivial and thus is not necessary.
we do not need docstring for every function/class/variable if the name is self-explainable

File: python/singa/autograd.py (hunk 1 above)
Comment (joddiy) [reply python/singa/autograd.py]: fixed

File: python/singa/autograd.py (hunk 2 above)
Comment (joddiy) [reply python/singa/autograd.py:4727]: fixed

File: python/singa/autograd.py (hunk 3 above)
Comment (joddiy) [reply python/singa/autograd.py]: fixed

**Pull request merged**